### DISCUSSION TITLE
Error in determining maximum BVAL

### DISCUSSION THREAD

**Discussion by markeckert** [muscbridge/PyDKE#303] [Technical Support] [answered]
Hi, I am using the Neurodock implementation of PyDesigner to process some older single shell data (b value = 700). PyDesigner processes the data through the eddy correction stage and then fails with "Error in determining maximum BVAL" (shown below).  The --verbose output suggests that PyDesigner is expecting at least a b value of 1000 (also shown below).  I created a dummy bval file replacing the 700 value with1000, which allowed PyDesigner to complete the processing and generate all of the typical output files (e.g., dti_dtistudio.fib). Is the higher b value dependency a bug or intentional?
Thanks for your help.
Mark 

Computing SNR: dwi_preprocessed.nii
   * B0...
   * B1000...
  
Image dwi_preprocessed.nii loaded successfully
Processing with 40 workers...
Traceback (most recent call last):
  File "/usr/local/bin/pydesigner", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/designer/pydesigner.py", line 980, in main
    nthreads=args.nthreads
  File "/usr/local/lib/python3.7/dist-packages/designer/fitting/dwipy.py", line 2694, in fit_regime
    protocols = img.tensorType()
  File "/usr/local/lib/python3.7/dist-packages/designer/fitting/dwipy.py", line 349, in tensorType
    raise ValueError('tensortype: Error in determining maximum '
ValueError: tensortype: Error in determining maximum BVAL

**Comment by TheJaeger** [comment]
@markeckert b-value=700 should work. The range for DTI is set from 500 to 1500, as defined in the [thresholds file](https://github.com/muscbridge/PyDesigner/blob/master/designer/fitting/thresholds.py). This is is a safeguard to prevent invalid datasets from being processed with PyDesigner. It's hard to say exactly what's causing this dataset to not be classified as DTI.

Would you mind sharing this dataset for me to look at?

**Reply by markeckert** [reply]
I am happy to upload an example (de-identified) dataset to an MUSC Box directory if you want to send me your email address to my MUSC email account.  Otherwise please let me know where I can upload the data. Thanks very much, Mark

**Reply by TheJaeger** [reply]
@markeckert I don't work at MUSC anymore but if you create a folder in MUSC box and share it with me at sid.kill3r@gmail.com (I know, I made it when I was 12) I can take a look.

**Comment by markeckert** [comment]
Hi Sid,
I have uploaded dcm and nii formatted diffusion data for you to look at. Please let me know if you have difficulty accessing the data. Also note that these de-identified data were collected at another institution and shared for a multi-site project. Please don’t share these data with anyone else.
Thanks!
Mark

**Reply by TheJaeger** [reply]
@markeckert I have received your Box invite. I'll be out this week, but will take a look at it when I get some time next week. I'll report my findings here.

**Reply by C-M-Patrick** [reply]
Any updates on this? I'm getting a similar error.

**Reply by C-M-Patrick** [reply, marked as answer]
Ah the maxBval function was interpreting b-values as INT, which caused values under 1000 to be rounded to 0. Updating the function to return a float resolved the issue
